app/javascript/src/components/Dashboard/Notes/NewNoteForm.jsx

```diff
@@ -30,7 +36,45 @@ export default function NewNoteForm({ onClose, refetch }) {
       {({ isSubmitting }) => (
         <Form>
           <Input label="Title" name="title" className="mb-6" />
-          <Textarea label="Description" name="description" rows={8} />
+          <Select
+            label="Tags"
+            options={tags}
+            name="tag"
+            className="mb-6"
+            placeholder="Select tag"
+          />
+          <Textarea
+            label="Note Description"
+            name="description"
+            rows={8}
+            className="mb-6"
+          />
+          <Select
+            label="Assigned Contact"
+            options={contacts}
+            name="assignedContact"
+            className="mb-6"
+            placeholder="Select contact"
+          />
+          <div className="mb-6">
+            Add Due Date to Note
+            <Switch
+              id="dueDateSwitch"
+              className="float-right"
+              checked={dueDateEnabled}
+              onChange={e => setDueDateEnabled(e.target.checked)}
+            />
+          </div>
+          {dueDateEnabled && (
+            <DateInput
+              type="date"
+              label="Due date"
+              name="dueDate"
+              format="DD/MM/YYYY"
+              value={dueDate}
+              onChange={setDueDate}
+            />
+          )}
```

Comment:
Rather than adding `mb-6` to each of the blocks individually, please use `space-y-6`.

Refer [space between](https://tailwindcss.com/docs/space#class-reference) from tailwindCSS for more details.